src/components/course/data/service.js

```diff
@@ -130,4 +141,42 @@ export default class CourseService {
     const url = `${this.config.LICENSE_MANAGER_URL}/api/v1/license-subsidy/?${qs.stringify(options)}`;
     return this.cachedAuthenticatedHttpClient.get(url);
   }
+
+  /**
+   * @typedef {Object} Offer An offer for a course
+   * @property {string} usageType
+   * @property {number} benefitValue
+   * @property {string} couponStartDate utc formatted
+   * @property {string} couponEndDate utc formatted
+   * @property {number} redemptionsRemaining
+   * @property {string} code
+   * @property {string} catalog uuid of catalog
+   */
```

Comment:
nit: is this jsdoc block needed?

Reply:
ah good question, I am using this typedef in the next jsdoc of the method to make it's output format clear. I think we could level this typedef up into a better place and reuse that typedef across the codebase (thought there was some extra step to make that happen so did not venture into it in this pr)